x-pack/plugins/apm/public/hooks/use_kuery_with_mobile_filters.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import {
+  HOST_OS_VERSION,
+  DEVICE_MODEL_NAME,
+  NETWORK_CONNECTION_TYPE,
+  SERVICE_VERSION,
+} from '../../common/es_fields/apm';
+import { paramQuery } from '../../common/utils/param_query';
+
+export function useKueryWithMobileFilters({
```

Comment:
why is this a hook? it returns a string, so memoization doesn't need to happen at this level I think. regenerating an object every time would invalidate a hook's dependencies but for strings that is not the case.